sdk/data/azcosmos/cosmos_item_request_patch_options.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcosmos
+
+// PatchOperationType defines supported values for operation types in Patch Document.
+type PatchOperationType string
+
+const (
+	// Represents a patch operation Add.
+	PatchOperationTypeAdd PatchOperationType = "add"
+	// Represents a patch operation Replace.
+	PatchOperationTypeReplace PatchOperationType = "replace"
+	// Represents a patch operation Remove.
+	PatchOperationTypeRemove PatchOperationType = "remove"
+	// Represents a patch operation Set.
+	PatchOperationTypeSet PatchOperationType = "set"
+	// Represents a patch operation Increment.
+	PatchOperationTypeIncrement PatchOperationType = "incr"
+)
+
+// PatchOperation represents individual patch operation.
+type PatchOperation struct {
+	Op    PatchOperationType `json:"op"`
+	Path  string             `json:"path"`
+	Value interface{}        `json:"value,omitempty"`
+}
+
+// PatchOptions represents the patch request.
+type PatchOptions struct {
+	Condition  *string          `json:"condition,omitempty"`
+	Operations []PatchOperation `json:"operations"`
+}
+
+// CreatePatchOptions returns a struct that represents the patch request.
+func CreatePatchOptions() *PatchOptions {
+	return &PatchOptions{
+		Condition:  nil,
+		Operations: make([]PatchOperation, 0),
+	}
+}
+
+// SetCondition sets condition for the patch request.
+func (p *PatchOptions) SetCondition(condition string) {
+	p.Condition = &condition
+}
+
+// Replace appends a replace operation to the patch request.
+func (p *PatchOptions) Replace(path string, value interface{}) {
```

Comment:
If we are adding public methods to add operations, should `Operations` be public? And in that line of thought, should the consts like `PatchOperationTypeReplace` or the struct `PatchOperation` also be public?

Otherwise this has two venues, users can manually add PatchOperations to the Operations array or use these methods, if we want them to use the methods, then the other alternative should be removed by making them private?